### DISCUSSION TITLE
Getting started with contributing

### DISCUSSION THREAD

**Discussion by andrewsi-z** [onnx/tensorflow-onnx#1695] [General]
Hello,

A bit of a general question here. For those that would like to contribute, there is a bit of a steep learning curve on some concepts here - primarily I am thinking about understanding the tensorflow graph and related info (such as keras layers), as well as how tf2onnx works with these.

Clearly one way to learn is to jump right in and begin experimenting...

I'd like to understand though if there are any "jump start" references that can be used to help get up to speed faster. Things like an overview of tf2onnx, or any info on tensorflow saved_model format (from community experiences) would be incredibly helpful.  
Does anything like this exist? I have looked, but have not found it yet. Are there other useful references anyone has found?

**Comment by guschmue** [comment]
A good start might be here:
https://www.tensorflow.org/guide/intro_to_graphs
A lot of the tensorflow documentation seems more targeted towards users and does not have a lot of details for developers.
In those case where you badly need to understand the details we find the tensorflow source helpful (for example there is little doc on the various tensorflow formats but their protobuf definitions tell all). 

tf2onnx always works on the graph level (doesn't matter if tf1, tf2 or keras) so that is the main thing that matters.
They can get pretty complex and we use https://github.com/lutzroeder/netron a lot to understand the graph.
